medusa/helpers/anidb.py

```diff
@@ -50,10 +50,10 @@ def set_up_anidb_connection():
 
 def create_key(namespace, fn, **kw):
     def generate_key(*args, **kw):
-        if six.PY3:
-            return namespace + '|' + args[0]
-        else:
-            return namespace + '|' + args[0].encode('utf-8')
+        show_key = namespace + '|' + args[0]
+        if six.PY2:
```

Comment:
Two things to consider here.  First having the python version check in the function requires it to be checked on every function call.  If there was a Python 3 version of the function and a Python 2 version of the function, you could check it a single time on import.  e.g.:

```python
def generate_key(*args, **kwargs):
    # Py 3 code here

def generate_key_py2(*args, **kwargs):
    # Py 2 code here

if six.PY2:
    generate_key = generate_key_py2
```
Additionally, since these only differ in their return type, you could make a simple wrapper that encodes when using Py2 and wrap the function with it.
```python
@py2encoded
def generate_key(*args, **kwargs):
    return namespace + '|' + args[0]
```
An example function wrapper would be as follows:
```python
import functools
import six

class Encoder:
    def __init__(self, encoding):
        self.encoding = encoding

    if six.PY2:
        def __call__(self, func):
            @functools.wraps(func)
            def wrapped_func(*args, **kwargs):
                return func(*args, **kwargs).encode(self.encoding)
            return wrapped_func
    else:
        def __call__(self, func):
            @functools.wraps(func)
            def wrapped_func(*args, **kwargs):
                return func(*args, **kwargs)
            return wrapped_func

py2encoded = Encoder('utf-8')
```

Reply:
Thank for the input @labrys. I think a decorator is a bit too much for this single case. Your other suggestion doesn't seem to be working for our code:
```
def create_key(namespace, fn, **kw):
    def generate_key(*args, **kw):
        show_key = namespace + '|' + args[0]
        if six.PY2:
            return show_key.encode('utf-8')
        return show_key
    return generate_key
```
Can you post the full code you had in mind?